akka-actor-tests/src/test/scala/akka/dispatch/VirtualThreadDispatcherSpec.scala

```diff
@@ -122,6 +125,50 @@ class VirtualThreadDispatcherSpec extends AnyWordSpec with Matchers {
         TestKit.shutdownActorSystem(system)
       }
     }
+
+    "nested works as expected" in {
+      if (JavaVersion.majorVersion < 21) {
+        // loom not available yet here
+        pending
+      } else {
+        implicit val system: ActorSystem = ActorSystem(
+          classOf[VirtualThreadDispatcherSpec].getSimpleName,
+          ConfigFactory.parseString("""
+              my-vt-dispatcher {
+                type = "Dispatcher"
+                executor = virtual-thread-executor
+                virtual-thread-executor {
+                  fallback="fork-join-executor"
+                }
+              }
+            """).withFallback(ConfigFactory.load()))
+
+        try {
+          implicit val dispatcher: ExecutionContext = system.dispatchers.lookup("my-vt-dispatcher")
+          val threadInfoProbe = TestProbe()
+          // only available on JDK 19+ so we can't use without reflection
+          def currentThreadId(): Long =
+            classOf[Thread].getMethod("threadId").invoke(Thread.currentThread()).asInstanceOf[Long]
+
+          Future {
+            threadInfoProbe.ref ! (("parent before", currentThreadId()))
+            Future {
+              threadInfoProbe.ref ! (("child before", currentThreadId()))
+              Thread.sleep(200)
+              threadInfoProbe.ref ! (("child after", currentThreadId()))
+            }
+            Thread.sleep(20)
+            threadInfoProbe.ref ! (("parent after", currentThreadId()))
+          }
+          val all = threadInfoProbe.receiveN(4)
+          val items = all.collect { case (evt: String, threadId: Long) => (evt, threadId) }
+          items.map(_._2).toSet should have size (2) // two different virtual threads
+          items.map(_._1) shouldEqual (Seq("parent before", "child before", "parent after", "child after"))
```

Comment:
This looks like it can be flaky. It might take more than 20 millis for the child to be scheduled and then "parent after" will come before "child before"

Reply:
Yeah, hmm, I couldn't figure out a way to make it completely deterministic without also making the reproducer go a way (like cross future synchronization something). Should I just tune the timeouts up a bit, I had them at 100 - 400 to start with but tuned down.